go/vt/vtexplain/vtexplain_vttablet.go

```diff
@@ -618,7 +618,7 @@ func (t *explainTablet) handleSelect(query string) (*sqltypes.Result, error) {
 
 	// Gen4 supports more complex queries so we now need to
 	// handle multiple FROM clauses
-	tables := make([]*sqlparser.AliasedTableExpr, len(selStmt.From))
+	tables := make([]*sqlparser.AliasedTableExpr, 0, len(selStmt.From))
```

Comment:
Found this bug as well while debugging. We create it with the length, which means subsequent `append` below would always create an even longer slice, negating this optimization plus having a lot of `nil` entries in this slice.

Reply:
It's really a `golang` design flaw. I've fallen for this many times.